opal/corelib/array.rb

```diff
@@ -63,7 +76,15 @@ def initialize(size = nil, obj = nil, &block)
   end
 
   def self.new(*args, &block)
-    [].initialize(*args, &block)
+    obj = allocate
+    breaker = obj.initialize(*args, &block)
+    %x{
+      if (obj.$$has_breaker !== undefined) {
```

Comment:
It's probably not, but at least in my case that seemed to be the least bad of the available options, until the way `break` works is fixed.